### ISSUE TITLE
Named windows

### ISSUE THREAD

**Issue by ggevay** [MaterializeInc/materialize#19755]
### Feature request

When the same OVER clause appears multiple times in a query, standard SQL allows us to simplify the query by specifying the OVER clause only once inside a WINDOW clause, naming it, and then referring to that name in the OVER clause:

```
SELECT
  *,
  RANK()       OVER w,
  DENSE_RANK() OVER w
  ROW_NUMBER() OVER w,
FROM table
WINDOW w AS (PARTITION BY user_id ORDER BY time);
```

The MySQL docs explain it nicely: https://dev.mysql.com/doc/refman/8.0/en/window-functions-named-windows.html

Postgres docs: https://dev.mysql.com/doc/refman/8.0/en/window-functions-named-windows.html (search for "WINDOW Clause".)

This is not urgent, we can get back to it when we have more users using window functions.

(Strictly speaking, this would be in Surfaces' domain, but probably I'll do it.)

**Comment by ggevay** [comment]
On a related note, we should also consider bundling together the computations of those window functions that have the same OVER clause, to reduce the overall overhead. I'll open a separate issue for this when we are further along with the prefix sum work. [An example user query where this would be useful.](https://materializeinc.slack.com/archives/C02PPB50ZHS/p1680558744782859)
